src/work-allocation-2/containers/work-case-list/work-case-list.component.ts

```diff
@@ -0,0 +1,225 @@
+import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import { Router } from '@angular/router';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { PaginationParameter } from 'src/work-allocation-2/models/dtos';
+
+import { ListConstants } from '../../components/constants';
+import { SortOrder } from '../../enums';
+import { Case, CaseAction, CaseServiceConfig, InvokedCaseAction } from '../../models/cases';
+import { FieldConfig, SortField } from '../../models/common';
+
+@Component({
+  selector: 'exui-work-case-list',
+  templateUrl: './work-case-list.component.html',
+  styleUrls: ['work-case-list.component.scss']
+})
+export class WorkCaseListComponent implements OnChanges {
+
+  /**
+   * These are the cases & fields as returned from the WA Api.
+   */
+  @Input() public cases: Case[];
+  @Input() public casesTotal: number;
+  @Input() public caseServiceConfig: CaseServiceConfig;
+  @Input() public sortedBy: SortField;
+  @Input() public addActionsColumn: boolean = true;
+  @Input() public pagination: PaginationParameter;
+  @Input() public showManage = {};
+
+  /**
+   * The message to display when there are no cases to display in the list.
+   */
+  @Input() public emptyMessage: string = ListConstants.EmptyMessage.Default;
+
+  // TODO: Need to re-read the LLD, but I believe it says pass in the caseServiceConfig into this CaseListComponent.
+  // Therefore we will not need this.
+  @Input() public fields: FieldConfig[];
+
+  @Output() public sortEvent = new EventEmitter<string>();
+  @Output() public paginationEvent = new EventEmitter<number>();
+  @Output() public actionEvent = new EventEmitter<InvokedCaseAction>();
+
+  /**
+   * The datasource is an Observable of data to be displayed, as per LLD.
+   */
+  public dataSource$: Observable<Case[]>;
+
+  public displayedColumns: string[];
+
+  private selectedCase: Case;
+
+  constructor(private readonly router: Router) {}
+
+  public get showResetSortButton(): boolean {
+    if (!this.sortedBy) {
+      return false;
+    }
+    const { defaultSortFieldName, defaultSortDirection } = this.caseServiceConfig;
+    if (this.sortedBy.fieldName === defaultSortFieldName && this.sortedBy.order === defaultSortDirection) {
+      return false;
+    }
+    return true;
+  }
+
+  public selectCaseFromUrlHash(url: string): Case | null {
```

Comment:
_Question:_
Does this need `| null`? `null` is perfectly acceptable to return if the return type is simply set to `Case`.

Reply:
null has been removed